src/lib/handler.rs

```diff
@@ -28,17 +58,30 @@ pub fn handle_rollback() -> Result<()> {
         }
         // Proceed with rollback if boot_counter is <= 0
         Some(counter) if counter <= 0 => {
-            log::info!("Greenboot will now attempt to rollback to previous deployment");
-            let status = Command::new("bootc")
-                .arg("rollback")
-                .status()
-                .context("Failed to execute bootc rollback")?;
-            if !status.success() {
-                bail!("Rollback error: {}", status);
+            log::info!("Greenboot will now attempt to rollback to a previous deployment.");
+            if let Some(deployment_cmd) = detect_os_deployment() {
+                log::info!(
+                    "Deployment manager '{}' detected, attempting rollback.",
+                    deployment_cmd
+                );
+                let status = Command::new(deployment_cmd)
+                    .arg("rollback")
+                    .status()
+                    .context(format!("Failed to execute '{} rollback'", deployment_cmd))?;
```

Comment:
If this command fails, the function will return a `Result` with the corresponding error. If not, the status would be "success" I believe, so I think the next sentence `if !status.sucesss()` will never be executed. Am I right or am I missing something?

Reply:
Answering myself: the `status()` function will try to run the comment and fail only if it's not able to execute it (maybe because the file is not found, it has no execution permissions.. etc). It will not return an error in case the command is executed, independently of the returned status code.